### PR TITLE
chore: Add wasm_memory_threshold field to DefiniteCanisterSettings

### PR DESCRIPTION
**Pull request by dragoljub-duric**
This PR adds a missing `wasm_memory_threshold` field to `DefiniteCanisterSettings`.

### CODE REVIEW

File: rs/nns/handlers/root/interface/src/client.rs [hunk 1]
```diff
@@ -220,6 +220,7 @@ impl SpyNnsRootCanisterClientReply {
                 reserved_cycles_limit: Some(candid::Nat::from(10_u32)),
                 wasm_memory_limit: Some(candid::Nat::from(11_u32)),
                 log_visibility: Some(LogVisibility::Controllers),
+                wasm_memory_threshold: Some(candid::Nat::from(0_u32)),
```
Comment (aterga) [inline rs/nns/handlers/root/interface/src/client.rs:223]: Let's add a comment explaining the semantics of `wasm_memory_threshold` being zero?

File: rs/nervous_system/clients/src/canister_status.rs [hunk 2]
```diff
@@ -66,6 +66,7 @@ pub struct DefiniteCanisterSettings {
     pub reserved_cycles_limit: Option<candid::Nat>,
     pub wasm_memory_limit: Option<candid::Nat>,
     pub log_visibility: Option<LogVisibility>,
+    pub wasm_memory_threshold: Option<candid::Nat>,
```
Comment (aterga) [inline rs/nervous_system/clients/src/canister_status.rs:69]: We now have a few limits / thresholds in these settings; would it make sense to add a doc string to explain the difference / link to the interface spec?

**Review by dsarlis — APPROVED** [review]

**Review by aterga — APPROVED** [review]
Approving the code changes, but please add some docs to the code, as mentioned.

**Review by mraszyk — APPROVED** [review]

**Comment by dragoljub-duric** [comment]
@aterga 
The tests are failing because it is necessary to update `candid` files on `sns` and `nns` canisters to reflect changes:
`type canister_settings = record {
    controllers : opt vec principal;
    compute_allocation : opt nat;
    memory_allocation : opt nat;
    freezing_threshold : opt nat;
    reserved_cycles_limit : opt nat;
    log_visibility : opt log_visibility;
    wasm_memory_limit : opt nat;
    wasm_memory_threshold : opt nat;
};

type definite_canister_settings = record {
    controllers : vec principal;
    compute_allocation : nat;
    memory_allocation : nat;
    freezing_threshold : nat;
    reserved_cycles_limit : nat;
    log_visibility : log_visibility;
    wasm_memory_limit : nat;
    wasm_memory_threshold: nat;`
I tried to do that but I was not able to finish it. Can you or someone from your team try to make that change, please?

File: rs/nns/handlers/root/interface/src/client.rs (hunk 1 above)
Comment (dragoljub-duric) [reply rs/nns/handlers/root/interface/src/client.rs:225]: Done.

File: rs/nervous_system/clients/src/canister_status.rs (hunk 2 above)
Comment (dragoljub-duric) [reply rs/nervous_system/clients/src/canister_status.rs:74]: That makes sense, I added a comment.

**Comment by dragoljub-duric** [comment]
@aterga @dsarlis Please take another look, I refactored the code to go around external dependency on `ic_management_canister_types::DefiniteCanisterSettingsArgs` in `sns`.

**Review by aterga — APPROVED** [review]

**Pull request merged**